packages/mui-core/src/AutocompleteUnstyled/useAutocomplete.js

```diff
@@ -945,8 +945,19 @@ export default function useAutocomplete(props) {
   };
 
   const handleInputMouseDown = (event) => {
-    if (inputValue === '' || !open) {
-      handlePopupIndicator(event);
+    const className = event.target.className;
+    // In order to have the listbox open by clicking anywhere on the `InputBase` div container,
+    // an `onMouseDown` listener is now attached to the `InputBase` div container.
+    // Previously, an `onMouseDown` listener was attached to the <input /> element inside the `InputBase` div container.
+    // Since other handlers (e.g., handlePopupIndicator()) already exist to handle clicking specific elements inside the container,
+    // this handler performs the following conditional check to ensure that it runs only when <input /> or the `InputBase` div container is clicked
+    if (
+      typeof className === 'string' &&
+      (className.includes('InputBase-root') || className.includes('InputBase-input'))
```

Comment:
I didn't add `Mui` because prefixes of classname are modifiable by the user now.